src/CSModule.sol

```diff
@@ -677,45 +677,55 @@ contract CSModule is
     }
 
     /// @inheritdoc ICSModule
-    function submitWithdrawal(
-        uint256 nodeOperatorId,
-        uint256 keyIndex,
-        uint256 amount,
-        bool isSlashed
+    function submitWithdrawals(
+        ValidatorWithdrawalInfo[] calldata withdrawalsInfo
     ) external onlyRole(VERIFIER_ROLE) {
-        _onlyExistingNodeOperator(nodeOperatorId);
-        NodeOperator storage no = _nodeOperators[nodeOperatorId];
-        if (keyIndex >= no.totalDepositedKeys) {
-            revert SigningKeysInvalidOffset();
-        }
+        for (uint256 i; i < withdrawalsInfo.length; ++i) {
+            uint256 nodeOperatorId = withdrawalsInfo[i].nodeOperatorId;
 
-        uint256 pointer = _keyPointer(nodeOperatorId, keyIndex);
-        if (_isValidatorWithdrawn[pointer]) revert AlreadyWithdrawn();
-
-        _isValidatorWithdrawn[pointer] = true;
-        unchecked {
-            ++no.totalWithdrawnKeys;
-        }
+            _onlyExistingNodeOperator(nodeOperatorId);
+            NodeOperator storage no = _nodeOperators[nodeOperatorId];
 
-        bytes memory pubkey = SigningKeys.loadKeys(nodeOperatorId, keyIndex, 1);
-        emit WithdrawalSubmitted(nodeOperatorId, keyIndex, amount, pubkey);
+            uint256 keyIndex = withdrawalsInfo[i].keyIndex;
+            if (keyIndex >= no.totalDepositedKeys) {
+                revert SigningKeysInvalidOffset();
+            }
 
-        if (isSlashed) {
-            // Bond curve should be reset to default in case of slashing. See https://hackmd.io/@lido/SygBLW5ja
-            accounting.resetBondCurve(nodeOperatorId);
-        }
+            uint256 pointer = _keyPointer(nodeOperatorId, keyIndex);
+            if (_isValidatorWithdrawn[pointer]) revert AlreadyWithdrawn();
 
-        if (DEPOSIT_SIZE > amount) {
+            _isValidatorWithdrawn[pointer] = true;
             unchecked {
-                accounting.penalize(nodeOperatorId, DEPOSIT_SIZE - amount);
+                ++no.totalWithdrawnKeys;
             }
-        }
 
-        // Nonce should be updated if depositableValidators change
-        _updateDepositableValidatorsCount({
-            nodeOperatorId: nodeOperatorId,
-            incrementNonceIfUpdated: true
-        });
+            bytes memory pubkey = SigningKeys.loadKeys(
+                nodeOperatorId,
+                keyIndex,
+                1
+            );
+
+            uint256 amount = withdrawalsInfo[i].amount;
```

Comment:
Read `withdrawalsInfo[i]` once at the beginning of the cycle

Reply:
Done